zeppelin-web/src/app/interpreter/interpreter.controller.js

```diff
@@ -723,16 +723,21 @@ function InterpreterCtrl ($rootScope, $scope, $http, baseUrlSrv, ngToast, $timeo
   }
 
   $scope.showSparkUI = function (settingId) {
-    $http.get(baseUrlSrv.getRestApiBase() + '/interpreter/getmetainfos/' + settingId + '?propName=url')
+    $http.get(baseUrlSrv.getRestApiBase() + '/interpreter/getmetainfos/' + settingId)
       .success(function (data, status, headers, config) {
-        let url = data.body.url
-        if (!url) {
-          BootstrapDialog.alert({
-            message: 'No spark application running'
-          })
+        if (data.body === undefined || !data.body.url) {
+          if (data.body !== undefined && data.body.sparkUiEnabled === 'false') {
```

Comment:
Good idea. Refactored code so we only have url and message. More generic no doubt.